protocol/contracts/beanstalk/init/InitDistribution.sol

```diff
@@ -0,0 +1,55 @@
+/*
+ SPDX-License-Identifier: MIT
+*/
+
+pragma solidity ^0.8.20;
+
+import {C} from "contracts/C.sol";
+import {AppStorage, Storage} from "contracts/beanstalk/AppStorage.sol";
+import {ShipmentPlanner} from "contracts/ecosystem/ShipmentPlanner.sol";
+
+// NOTE: Values are arbitrary placeholders.
+
+interface IBeanstalk {
+    function setShipmentRoutes(Storage.ShipmentRoute[] calldata shipmentRoutes) external;
+}
+
+/**
+ * @author funderbrker
+ * @notice Initializes logistics and shipment routes.
+ */
+contract InitDistribution {
+    address constant BEANSTALK = 0xC1E088fC1323b20BCBee9bd1B9fC9546db5624C5;
+
+    AppStorage internal s;
+    IBeanstalk beanstalk = IBeanstalk(BEANSTALK);
+
+    address shipmentPlanner = address(new ShipmentPlanner());
+
+    function init() external {
+        Storage.ShipmentRoute[] memory shipmentRoutes = new Storage.ShipmentRoute[](3);
+
+        shipmentRoutes[0] = Storage.ShipmentRoute(
+            shipmentPlanner,
+            ShipmentPlanner.getSiloPlan.selector,
+            Storage.Recipient.Silo,
+            bytes("")
+        );
+
+        shipmentRoutes[1] = Storage.ShipmentRoute(
+            shipmentPlanner,
+            ShipmentPlanner.getFieldPlan.selector,
+            Storage.Recipient.Field,
+            bytes("")
+        );
+
+        shipmentRoutes[2] = Storage.ShipmentRoute(
+            shipmentPlanner,
+            ShipmentPlanner.getBarnPlan.selector,
+            Storage.Recipient.Barn,
+            bytes("")
+        );
+
+        BEANSTALK.delegatecall(abi.encodeWithSelector(beanstalk.setShipmentRoutes.selector, shipmentRoutes));
```

Comment:
Can we call the function directly rather than delegate calling?

Also, while we can use the beanstalk address directly, the benefit of using `IBeanstalk(address.this)` is that any beanstalk fork can initialize their beanstalk using the existing init contract deployed rather than having to redeploy their own `InitDistribution.sol`.
```suggestion
        IBeanstalk(address(this)).setShipmentRoutes(shipmentRoutes);
```

Reply:
setShipmetRoutes is an external function that is only accessible to the owner. So using delegate call keeps the msg.sender address as the owner. 

Would it be better to set it to only be external but only callable by Beanstalk address such that it can only be called through Inits and DiamondCuts?

Are there any examples of other external functions that are protected similarly?

Reply:
Got it. Totally slipped my mind re msg.sender. In cases where an external function is access locked, we've historically placed the logic in a library (where the external function is a wrapper around it + access control). Then in the init contract we call the library. This definitely works as well unchanged.

Reply:
partially achieved here, although i was unable to call the beanstalk contract object directly 

https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/d431b0731dbf151a4f810b4e7bdfab1a728a323f#diff-b0451bea09abb2293197eb00e388806cf4949fc7fd2952a7c926693780a93f60R56-R57

Reply:
got it
https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/d5c6b3cc0e9749cb496e1de9a924392da46cd8c4#diff-b0451bea09abb2293197eb00e388806cf4949fc7fd2952a7c926693780a93f60R60-R62

Reply:
lgtm